sql/schemachanger: rework datadriven output

ajwerner opened the pull request:
This PR first redacts most of the overly-noisy declarative schema changer state from the SideEffects output, and then adds explain diagrams for each of the tests in the second commit as separate files, lastly it refactors some shared logic to make it straightforward to remove the embedded fields everywhere.

Release note: None

ajwerner committed:
sql/schemachanger: redact declarative schema changer state in side-effects

It's too noisy.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/84469)
<!-- Reviewable:end -->

ajwerner committed:
sql/schemachanger: add explain diagrams to datadriven output

Now that we have just one test directive per file, it's clear how to map new
files to individual test directives. This work also assumes that multi-statement
test directives are coming.

Release note: None

ajwerner committed:
sql/schemachanger/scplan/scviz: extract logic, lift embedded fields

This commit adds a new package which allows scgraphviz and sctestutils to share
code. The end result is that the embedded struct fields can be hoisted into
their parent for explain output.

Release note: None

postamar reviewed (approved):
Yes please! This is much, much nicer.

This is the only thing I have to complain about, and it can be done as a followup: please remove the `EXPLAIN` test cases from the `new_schema_changer` logictest suite and perhaps port them to `explain` if they have value. They bloat the logic test definition quite unreasonably.

ajwerner commented:
TFTR!

bors r+
